specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5214 @@
+        },
+        "properties": {
+          "$ref": "#/definitions/LinkedWorkspaceProps"
+        }
+      }
+    },
+    "LinkedWorkspaceDto": {
+      "type": "object",
+      "description": "object used for creating linked workspace.",
+      "properties": {
+        "name": {
+          "description": "Friendly name of the linked workspace",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/LinkedWorkspaceProps"
+        }
+      }
+    },
+    "LinkedWorkspaceProps": {
+      "type": "object",
+      "description": "LinkedWorkspace specific properties.",
+      "properties": {
+        "linkedWorkspaceResourceId": {
+          "description": "ResourceId of the link target of the linked workspace.",
+          "type": "string"
+        },
+        "userAssignedIdentityResourceId": {
+          "description": "ResourceId of the user assigned identity for the linked workspace.",
+          "type": "string"
+        }
+      }
+    },
+    "ServiceResource": {
+      "type": "object",
+      "description": "Machine Learning service object wrapped into ARM resource envelope.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "description": "Service properties",
+              "$ref": "#/definitions/ServiceResponseBase"
+            }
+          }
+        }
+      ]
+    },
+    "ServiceResponseBase": {
+      "description": "The base service response. The correct inherited response based on computeType will be returned (ex. ACIServiceResponse)",
+      "required": [
+        "computeType"
+      ],
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "The service description.",
+          "type": "string"
+        },
+        "kvTags": {
+          "description": "The service tag dictionary. Tags are mutable.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "description": "The service property dictionary. Properties are immutable.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "operationId": {
+          "description": "The ID of the latest asynchronous operation for this service.",
+          "type": "string"
+        },
+        "state": {
+          "description": "The current state of the service.",
+          "enum": [
+            "Transitioning",
+            "Healthy",
+            "Unhealthy",
+            "Failed",
+            "Unschedulable"
+          ],
+          "type": "string",
+          "example": "Healthy",
+          "x-ms-enum": {
+            "name": "WebServiceState",
+            "modelAsString": false
+          }
+        },
+        "createdTime": {
```

Comment:
these still need to be removed or marked readonly.  Please mark anything in here that is readonly as readonly

Reply:
removed. marked other readonly properties as readonly.